src/Blockcore.Indexer.Core/Handlers/CommandHandler.cs

```diff
@@ -31,9 +38,45 @@ public async Task<string> SendTransaction(string transactionHex)
          // todo: check how a failure is porpageted
 
          SyncConnection connection = syncConnection;
-         var client = clientFactory.Create(connection);
-         string trxid = await client.SentRawTransactionAsync(transactionHex);
-         return trxid;
+         Transaction trx = null;
+         try
+         {
+            // parse the trx;
+            trx = connection.Network.Consensus.ConsensusFactory.CreateTransaction(transactionHex);
+            trx.PrecomputeHash(false, true);
+
+         }
+         catch (Exception ex)
+         {
+            log.LogDebug($"Invalid trx received {ex.Message}");
+            return string.Empty;
+         }
+
+         if (trx == null)
+            return string.Empty;
+
+         try
+         {
+            IBlockchainClient client = clientFactory.Create(connection);
+            string trxid = await client.SentRawTransactionAsync(transactionHex);
```

Comment:
One thing that I wonder here is perhaps I should fetch the trx hex from the node itself to be sure it was added properly, though I think the send will already do that.